pkgs/applications/misc/navit/default.nix

```diff
@@ -68,10 +68,11 @@ stdenv.mkDerivation rec {
   '';
 
   # TODO: fix upstream?
-  postFixup = ''
-    for lib in $(find "$out/lib/navit/" -iname "*.so" ); do
-      patchelf --set-rpath ${makeLibraryPath buildInputs} $lib
-    done
+  libPath = stdenv.lib.makeLibraryPath ([ stdenv.cc.libc ] ++ buildInputs );
+  postFixup =
+  ''
+	  find "$out/lib" -type f -name "*.so" -exec patchelf --set-rpath $libPath {} \;
```

Comment:
Have you tried `autoPatchelfHook` in `nativeBuildInputs`? I found it usually more reliable then manual patchelf.

Reply:
autoPatchelfHook doesn't fix glibc missing library.